jdbc/src/ragtime/jdbc.clj

```diff
@@ -35,32 +35,46 @@
                        (str (:table_schem metadata) "." (:table_name metadata))
                        (:table_name metadata))))
 
-(defn- table-exists? [db-spec ^String table-name]
+(defn- table-exists-via-metadata-scan? [db-spec ^String table-name]
   (some (partial metadata-matches-table? table-name)
         (get-table-metadata db-spec)))
 
-(defn- ensure-migrations-table-exists [db-spec migrations-table]
-  (when-not (table-exists? db-spec migrations-table)
+(defn- table-exists-via-provided-sql? [db-spec ^String migrations-table-exists-sql]
+  (let [row-count (count (sql/query db-spec [migrations-table-exists-sql]))]
+    (case row-count
+      0 false
+      1 true
+      (throw (ex-info "Provided SQL doesn't directly answer if a migration table exists or not"
```

Comment:
Can you ensure all lines are 80 characters or fewer. Also why throw an error if the row count is greater than 1? Why not just have it be false for zero rows, true otherwise?